### PR TITLE
Add pysbd sentence boundary predictor

### PR DESCRIPTION
**Pull request by lolipopshock**
WIP -- need to add tests

### CODE REVIEW

File: mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                or char2token_mask[end + 1] != token_id_end
+            ):
+                token_id_end += 1  # (Including the end)
+            split.append((token_id_start, token_id_end))
+            token_id_start = token_id_end
+        return split
+
+    def predict(self, doc: Document) -> List[SpanGroup]:
+
+        words = [token.symbols[0] for token in doc.tokens]
```
Comment (lolipopshock) [inline mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py:73]: # TODO: probably `words` is in incorrect reading order.

File: mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        sentence_spans = []
+        for (start, end) in split:
+            if end - start == 0:
+                continue
+            if end - start < 0:
+                raise ValueError
+            cur_spans = doc.tokens[start:end]
+            start = min([ele.start for ele in cur_spans])
+            end = max([ele.end for ele in cur_spans])
```
Comment (lolipopshock) [inline mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py:84]: change to Union

Comment (lolipopshock) [reply mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py:84]: merge_neighbors

File: mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            if end - start < 0:
+                raise ValueError
+
+            cur_spans = getattr(doc, attr_name)[start:end]
+
+            all_token_spans = list(
+                itertools.chain.from_iterable([ele.spans for ele in cur_spans])
+            )
+
+            sentence_spans.append(SpanGroup(merge_neighbor_spans(all_token_spans)))
```
Comment (lolipopshock) [inline mmda/predictors/heuristic_predictors/sentence_boundary_predictor.py:146]: @kyleclo

**Review by kyleclo — APPROVED** [review]
lets try it out

**Pull request merged**